plugins/tiledb/test/TileDBWriterTest.cpp

```diff
@@ -0,0 +1,102 @@
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Consulting LLC nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include <stdio.h>
+#include <sys/types.h>
+#include <dirent.h>
+
+#include <pdal/pdal_test_main.hpp>
+#include <io/FauxReader.hpp>
+#include <pdal/StageFactory.hpp>
+
+#include "Support.hpp"
+#include "../io/TileDBWriter.hpp"
+
+namespace pdal
+{
+    class TileDBWriterTest : public ::testing::Test
+    {
+    protected:
+        virtual void SetUp()
+        {
+            Options options;
+            options.add("mode", "ramp");
+            options.add("count", 100);
+            m_reader.setOptions(options);
+        }
+
+        FauxReader m_reader;
+
+    };
+
+    TEST_F(TileDBWriterTest, constructor)
+    {
+        TileDBWriter writer;
+    }
+
+    TEST_F(TileDBWriterTest, findStage)
+    {
+        StageFactory factory;
+        Stage* stage(factory.createStage("writers.tiledb"));
+        EXPECT_TRUE(stage);
+    }
+
+    TEST_F(TileDBWriterTest, write)
+    {
+        tiledb::Context ctx;
+        tiledb::VFS vfs(ctx);
+
+        Options options;
+        std::string pth = Support::temppath("tiledb_test_out");
+        std::string sidecar = pth + "/pdal.json";
+        options.add("array_name", pth);
+
+        if (vfs.is_dir(pth))
+        {
+            vfs.remove_dir(pth);
+        }
+
+        TileDBWriter writer;
+        writer.setOptions(options);
+        writer.setInput(m_reader);
+
+        PointTable table;
+        writer.prepare(table);
+        writer.execute(table);
+
+        // check the sidecar exists
+        EXPECT_TRUE(pdal::Utils::fileExists(sidecar));
+    }
+}
```

Comment:
It would be nice if the tests made sure that data was written as expected and read as expected.

Reply:
I will add more to the tests.